### PR TITLE
feature/print op

### PR DESCRIPTION
**Pull request by ChunweiYan**
fixes: https://github.com/PaddlePaddle/Paddle/issues/6788

### CODE REVIEW

File: paddle/operators/CMakeLists.txt [hunk 1]
```diff
@@ -260,6 +261,7 @@ op_library(recurrent_op SRCS recurrent_op.cc DEPS executor)
 # FIXME(typhoonzero): save/load depends lodtensor serialization functions
 op_library(save_op DEPS lod_tensor)
 op_library(load_op DEPS lod_tensor)
+op_library(print_op SRCS print_op.cc DEPS lod_tensor)
```
Comment (jacquesqiao) [inline paddle/operators/CMakeLists.txt]: what kind of operator need to declare in this CMakeLists?

File: python/paddle/v2/fluid/layers/control_flow.py [hunk 2]
```diff
@@ -44,6 +62,48 @@ def merge_lod_tensor(in_true, in_false, x, mask, level=0):
...
+    '''
+    helper = LayerHelper('print', **locals())
+    out = helper.create_tmp_variable(dtype='int32')
+    helper.append_op(
+        type='print',
+        inputs={'input': input},
+        attrs={
+            'first_n': first_n,
+            'summarize': summarize,
+            'message': message if message else "",
```
Comment (jacquesqiao) [inline python/paddle/v2/fluid/layers/control_flow.py]: can use 
```python
message': message or ""
```

File: python/paddle/v2/fluid/layers/control_flow.py [hunk 3]
```diff
@@ -44,6 +62,48 @@ def merge_lod_tensor(in_true, in_false, x, mask, level=0):
...
+    '''
+    Creates a print op that will print when a tensor is accessed.
+
+    Wraps the tensor passed in so that whenever that a tensor is accessed,
+    the message `message` is printed, along with the current value of the
+    tensor `t`.
+
+    Args:
+      input: A Tensor to print.
+      summarize: Print this number of elements in the tensor.
```
Comment (qingqing01) [inline python/paddle/v2/fluid/layers/control_flow.py]: Should tell the user what will be happened when `summarize=-1`.

File: python/paddle/v2/fluid/layers/control_flow.py [hunk 4]
```diff
@@ -44,6 +62,48 @@ def merge_lod_tensor(in_true, in_false, x, mask, level=0):
...
+      message: A string message to print as a prefix.
+      first_n: Only log `first_n` number of times.
+      print_tensor_name: Print the tensor name.
+      print_tensor_type: Print the tensor type.
+      print_tensor_shape: Print the tensor shape.
+      print_tensor_lod: Print the tensor lod.
```
Comment (qingqing01) [inline python/paddle/v2/fluid/layers/control_flow.py]: The format of doc for Python API is here: https://github.com/PaddlePaddle/Paddle/issues/6526

> Print the tensor name
whether to print ...

File: python/paddle/v2/fluid/layers/control_flow.py [hunk 5]
```diff
@@ -44,6 +62,48 @@ def merge_lod_tensor(in_true, in_false, x, mask, level=0):
...
+
+    Wraps the tensor passed in so that whenever that a tensor is accessed,
+    the message `message` is printed, along with the current value of the
+    tensor `t`.
+
+    Args:
+      input: A Tensor to print.
+      summarize: Print this number of elements in the tensor.
+      message: A string message to print as a prefix.
+      first_n: Only log `first_n` number of times.
```
Comment (qingqing01) [inline python/paddle/v2/fluid/layers/control_flow.py]: > Only log `first_n` number of times

这个注释不是特别明白~

File: python/paddle/v2/fluid/layers/control_flow.py [hunk 6]
```diff
@@ -44,6 +62,48 @@ def merge_lod_tensor(in_true, in_false, x, mask, level=0):
...
+def Print(input,
+          first_n=-1,
+          message=None,
+          summarize=-1,
+          print_tensor_name=True,
+          print_tensor_type=True,
+          print_tensor_shape=True,
+          print_tensor_lod=True):
+    '''
+    Creates a print op that will print when a tensor is accessed.
```
Comment (abhinavarora) [inline python/paddle/v2/fluid/layers/control_flow.py]: Make sure that the layer comments are same as the convention defined in https://github.com/PaddlePaddle/Paddle/pull/6806

File: python/paddle/v2/fluid/layers/control_flow.py (hunk 5 above)
Comment (ChunweiYan) [reply python/paddle/v2/fluid/layers/control_flow.py]: 只有前 `frist_n` 次forward时会打log

File: python/paddle/v2/fluid/layers/control_flow.py (hunk 6 above)
Comment (ChunweiYan) [reply python/paddle/v2/fluid/layers/control_flow.py]: done

File: paddle/operators/print_op.cc [hunk 7]
```diff
@@ -0,0 +1,205 @@
...
+      Display<int>();
+    }
+    if (dtype.hash_code() == typeid(int64_t).hash_code()) {
+      Display<int64_t>();
+    }
+  }
+
+  template <typename T>
+  void Display() {
+    auto* d = (T*)data;
```
Comment (jacquesqiao) [inline paddle/operators/print_op.cc]: 如果是GPU tensor，这里可能会挂

File: paddle/operators/print_op.cc [hunk 8]
```diff
@@ -0,0 +1,205 @@
...
+  void Run(const framework::Scope& scope,
+           const platform::Place& place) const override {
+    // Only run the `first_n` times.
+    int first_n = Attr<int>("first_n");
+    if (first_n > 0 && ++times_ > first_n) return;
+
+    PADDLE_ENFORCE(!Inputs("input").empty(), "input should be set");
+    auto* input_var = scope.FindVar(Input("input"));
+    PADDLE_ENFORCE_NOT_NULL(input_var);
+    auto& tensor = input_var->Get<framework::LoDTensor>();
```
Comment (jacquesqiao) [inline paddle/operators/print_op.cc]: enforce(is_cpu_place(tensor.place()))

File: paddle/operators/print_op.cc [hunk 9]
```diff
@@ -0,0 +1,208 @@
...
+
+    // TODO(ChunweiYan) support GPU
+    PADDLE_ENFORCE(framework::is_cpu_place(tensor.place()));
```
Comment (pkuyym) [inline paddle/operators/print_op.cc]: framework --> platform

**Pull request merged**